src/NUnitFramework/framework/Internal/TypeHelper.cs

```diff
@@ -250,25 +250,22 @@ public static void ConvertArgumentList(object[] arglist, IParameterInfo[] parame
             {
                 object arg = arglist[i];
 
-                if (arg != null && arg is IConvertible)
+                if (arg is IConvertible)
                 {
                     Type argType = arg.GetType();
                     Type targetType = parameters[i].ParameterType;
                     bool convert = false;
 
-                    if (argType != targetType && !argType.IsAssignableFrom(targetType))
```

Comment:
Can you clarify why !IsAssignableFrom() can be safely dropped here? Sorry - there's a lot going on in this method!

Reply:
Oh! Sorry. This would have been the benefit of making this change in its own PR to start with– there should have been a whole commit making just this change with a commit message explaining why.

The reason is that the only way `convert` can become true is if both types are considered numeric by `IsNumeric`. There are no such types that are assignable to one another except in the trivial case of `argType == targetType` which is already guarded against, so `argType.IsAssignableFrom(targetType)` is guaranteed to always be false when evaluated after the two `&& IsNumeric` checks.

Reply:
Oh, and I did a quick brute-force LINQPad test to make sure that my instinct was right about this. because that's an important thing to verify. 😄

Reply:
Great - thanks! 😄 Approving - just waiting on CI